# Housing Market

### Introduction:

This time we will create our own dataset with fictional numbers to describe a house market. As we are going to create random data don't try to reason of the numbers.

### Step 1. Import the necessary libraries

In [0]:
import numpy as np
import pandas as pd

### Step 2. Create 3 differents Series, each of length 100, as follows: 
1. The first a random number from 1 to 4 
2. The second a random number from 1 to 3
3. The third a random number from 10,000 to 30,000

In [0]:
rand_series1=np.random.randint(1,4, size=100)

In [3]:
rand_series1

array([3, 1, 1, 3, 2, 2, 2, 2, 2, 2, 2, 1, 3, 1, 3, 3, 1, 2, 3, 3, 1, 2,
       3, 1, 1, 1, 3, 2, 3, 1, 2, 2, 2, 3, 1, 3, 1, 1, 2, 2, 2, 1, 3, 2,
       2, 2, 1, 2, 2, 3, 3, 3, 2, 3, 3, 3, 3, 2, 1, 1, 2, 1, 3, 1, 3, 3,
       3, 2, 2, 1, 2, 2, 3, 2, 2, 2, 2, 3, 1, 2, 1, 3, 2, 2, 2, 2, 1, 3,
       3, 2, 3, 2, 1, 3, 2, 2, 3, 1, 2, 1])

In [0]:
rand_series2=np.random.randint(1,3, size=100)

In [5]:
rand_series2

array([2, 1, 1, 1, 1, 2, 1, 2, 2, 2, 1, 2, 1, 2, 1, 2, 2, 1, 2, 1, 2, 1,
       2, 1, 2, 2, 1, 2, 1, 2, 1, 1, 1, 1, 1, 2, 1, 2, 2, 2, 1, 1, 2, 2,
       2, 2, 1, 2, 2, 1, 1, 2, 2, 1, 1, 1, 1, 1, 2, 2, 2, 2, 1, 1, 1, 2,
       2, 1, 1, 2, 1, 2, 2, 2, 1, 1, 1, 2, 2, 1, 1, 1, 2, 2, 1, 1, 2, 2,
       2, 1, 1, 1, 1, 1, 2, 1, 2, 2, 1, 2])

In [0]:
rand_series3=np.random.randint(10000, 30000, size=100)

In [7]:
rand_series3

array([22310, 27534, 21565, 15488, 18805, 20938, 15717, 20024, 26747,
       19077, 13338, 15037, 29516, 16917, 12953, 24777, 27108, 16281,
       21341, 29848, 11245, 26908, 12381, 12602, 18595, 14160, 23244,
       11688, 23751, 19448, 16734, 14554, 28233, 29884, 19272, 12071,
       25439, 21048, 12986, 26348, 13440, 27612, 16546, 19932, 17077,
       25172, 16592, 24074, 11479, 15977, 25952, 11129, 21653, 23829,
       20503, 14658, 20996, 15769, 21318, 18547, 25925, 29240, 13813,
       23210, 26213, 17390, 16156, 12757, 26650, 24830, 28748, 21832,
       20381, 12783, 19737, 29816, 29167, 13254, 25061, 20261, 13376,
       16989, 24132, 21808, 16130, 18047, 17621, 17582, 22101, 22524,
       21335, 12639, 24024, 21612, 19834, 18357, 11430, 28547, 19239,
       19724])

### Step 3. Let's create a DataFrame by joinning the Series by column

In [0]:
rand_df=pd.DataFrame(zip(rand_series1,rand_series2,rand_series3), columns=['series1', 'series2', 'series3'])

In [9]:
rand_df.head()

series1  series2  series3
0        3        2    22310
1        1        1    27534
2        1        1    21565
3        3        1    15488
4        2        1    18805

### Step 4. Change the name of the columns to bedrs, bathrs, price_sqr_meter

In [0]:
rand_df.columns=['bedrs','bathrs','price_sqr_meter']

In [11]:
rand_df.head()

bedrs  bathrs  price_sqr_meter
0      3       2            22310
1      1       1            27534
2      1       1            21565
3      3       1            15488
4      2       1            18805

In [13]:
rand_df.tail()

bedrs  bathrs  price_sqr_meter
95      2       1            18357
96      3       2            11430
97      1       2            28547
98      2       1            19239
99      1       2            19724

### Step 5. Create a one column DataFrame with the values of the 3 Series and assign it to 'bigcolumn'

In [0]:
bigcolumn=pd.concat([pd.Series(rand_series1),pd.Series(rand_series2),pd.Series(rand_series3)], axis=0)

In [16]:
bigcolumn

0         3
1         1
2         1
3         3
4         2
5         2
6         2
7         2
8         2
9         2
10        2
11        1
12        3
13        1
14        3
15        3
16        1
17        2
18        3
19        3
20        1
21        2
22        3
23        1
24        1
25        1
26        3
27        2
28        3
29        1
      ...  
70    28748
71    21832
72    20381
73    12783
74    19737
75    29816
76    29167
77    13254
78    25061
79    20261
80    13376
81    16989
82    24132
83    21808
84    16130
85    18047
86    17621
87    17582
88    22101
89    22524
90    21335
91    12639
92    24024
93    21612
94    19834
95    18357
96    11430
97    28547
98    19239
99    19724
Length: 300, dtype: int64

In [0]:
big_df=pd.DataFrame(bigcolumn, columns=['bigseries'])

In [19]:
big_df

bigseries
0           3
1           1
2           1
3           3
4           2
5           2
6           2
7           2
8           2
9           2
10          2
11          1
12          3
13          1
14          3
15          3
16          1
17          2
18          3
19          3
20          1
21          2
22          3
23          1
24          1
25          1
26          3
27          2
28          3
29          1
..        ...
70      28748
71      21832
72      20381
73      12783
74      19737
75      29816
76      29167
77      13254
78      25061
79      20261
80      13376
81      16989
82      24132
83      21808
84      16130
85      18047
86      17621
87      17582
88      22101
89      22524
90      21335
91      12639
92      24024
93      21612
94      19834
95      18357
96      11430
97      28547
98      19239
99      19724

[300 rows x 1 columns]

### Step 6. Ops it seems it is going only until index 99. Is it true?

In [0]:
idx_series=pd.Series(np.arange(0,300,1))

In [22]:
idx_series.size

300

In [23]:
idx_series

0        0
1        1
2        2
3        3
4        4
5        5
6        6
7        7
8        8
9        9
10      10
11      11
12      12
13      13
14      14
15      15
16      16
17      17
18      18
19      19
20      20
21      21
22      22
23      23
24      24
25      25
26      26
27      27
28      28
29      29
      ... 
270    270
271    271
272    272
273    273
274    274
275    275
276    276
277    277
278    278
279    279
280    280
281    281
282    282
283    283
284    284
285    285
286    286
287    287
288    288
289    289
290    290
291    291
292    292
293    293
294    294
295    295
296    296
297    297
298    298
299    299
Length: 300, dtype: int64

In [0]:
big_df.set_index(idx_series,drop=True, inplace=True)

In [26]:
big_df.head()

bigseries
0          3
1          1
2          1
3          3
4          2

In [27]:
big_df.tail()

bigseries
295      18357
296      11430
297      28547
298      19239
299      19724

### Step 7. Reindex the DataFrame so it goes from 0 to 299

In [28]:
big_df

bigseries
0            3
1            1
2            1
3            3
4            2
5            2
6            2
7            2
8            2
9            2
10           2
11           1
12           3
13           1
14           3
15           3
16           1
17           2
18           3
19           3
20           1
21           2
22           3
23           1
24           1
25           1
26           3
27           2
28           3
29           1
..         ...
270      28748
271      21832
272      20381
273      12783
274      19737
275      29816
276      29167
277      13254
278      25061
279      20261
280      13376
281      16989
282      24132
283      21808
284      16130
285      18047
286      17621
287      17582
288      22101
289      22524
290      21335
291      12639
292      24024
293      21612
294      19834
295      18357
296      11430
297      28547
298      19239
299      19724

[300 rows x 1 columns]